Admin scripts

Allcharles opened the pull request:
# Admin Script Pages

Created the admin script pages

## Changes

- Admin Script Page
- Admin New Script Page
- `ScriptsService`

## Issues

- Did not create unit tests for `ScriptsService`, `AdminScriptsComponent`, and `AdminScriptsNewComponent`

## Closes

Closes #130 

## Visual Changes

Admin Scripts Page
![image](https://user-images.githubusercontent.com/3955116/76277622-fae32380-62d4-11ea-975f-c454a4ee3b2b.png)

Admin New Script Page
![image](https://user-images.githubusercontent.com/3955116/76277653-0b939980-62d5-11ea-83d5-2d3502c5cd87.png)

Allcharles committed:
Created basic scripts page

Allcharles committed:
Created scripts category

Allcharles committed:
Created new script page

Allcharles committed:
Linted project

atruskie commented on src/app/models/AbstractModel.ts:
```diff
@@ -15,6 +15,8 @@ export abstract class AbstractModel {
     return new _new(raw);
   }
+
+  public abstract redirectPath(...args: any): string;
```
what does this do?

Ass some docs?

Allcharles committed:
Added documentation to AbstractModel

Allcharles committed:
Cleanup ScriptService

- Created ScriptService resolvers
- Removed providedIn: "root" for scriptsservice
- Updated app providers
